tooling/v8-snapshot/src/doctor/snapshot-doctor.ts

```diff
@@ -44,8 +44,30 @@ export type SnapshotDoctorOpts = Omit<
 > & {
   previousDeferred: Set<string>
   previousHealthy: Set<string>
-  previousNoRewrite: Set<string>
-  forceNoRewrite: Set<string>
+  previousNorewrite: Set<string>
+  forceNorewrite: Set<string>
+}
+
+/**
+ * Checks if a dependency matches a force no rewrite entry
+ * @param dependency - The dependency to check
+ * @param forceNorewrite - The force no rewrite entry
+ * @returns true if the dependency matches the force no rewrite entry, false otherwise
+ */
+export const doesDependencyMatchForceNorewriteEntry = (dependency: string, forceNorewrite: string) => {
+  // The force no rewrite file follows a convention where we try
+  // and match all possible node_modules paths if the force no
+  // rewrite entry starts with "*". If it does not
+  // start with "*" then it is an exact match.
+  if (forceNorewrite.startsWith('*')) {
+    if (dependency.endsWith(forceNorewrite.slice(2))) {
```

Comment:
Why do we only check for `*` but `slice(2)`?

Reply:
Good catch. I think this was a remnant of an earlier thought. Still worked but checking for */ is more accurate: [`01200cd` (#31551)](https://github.com/cypress-io/cypress/pull/31551/commits/01200cdadbd5271e7439de5a2dafc5a8e2be9168)